### ISSUE TITLE
[Chip] When `closable` the component should not have nested elements

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#14127]
**monday.com sync:** #11552962386

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Chip

### Summary

When the component is `closable` the component's close button should not be nested and should pass the accessible test in our suite.

### Actual Behavior

The component's `chip` contains a nested close button when `closable`:

```html
<calcite-chip>
  <button>
    My chip
    <button>X</button>
  </button>
</calcite-chip>
```

### Expected Behavior

Markup should not be nested:

```html
<calcite-chip>
  <button>
    My chip
  </button>
  <button>X</button>
</calcite-chip>
```

### Reproduction Sample

- https://codepen.io/geospatialem/pen/MYjvWVp
- https://codepen.io/driskull/pen/QwKMwgq

### Reproduction Steps

_Can be reproduced locally using the accessible test_

> accessible(`<calcite-chip label="Gray basemap" icon="basemap" closable></calcite-chip>`);

### Reproduction Version

5.0.2

### Relevant Info

Our `chip`'s accessible test also fails:

> accessible(`<calcite-chip label="Gray basemap" icon="basemap" closable></calcite-chip>`);

Both a11yhawk and axe list the issue as [**serious**](https://dequeuniversity.com/rules/axe/4.10/nested-interactive?application=axeAPI), however funtionality-wise it doesn't seem to impact users on initial testing.

Refer to the internal repo `ArcGISDevelopers/arcgis-platform-site/issues/#1077` for more info

> ### Why it matters
> Focusable elements with an interactive control ancestor (any element that accepts user input such as button or anchor elements) are not announced by screen readers and create an empty tab stop. That is, you could tab to the element but the screen reader will not announce its name, role, or state.

<img width="1868" height="692" alt="Image" src="https://github.com/user-attachments/assets/8714c980-0bea-47f3-a191-9c77e587d39f" />

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Developer Experience

**Comment by driskull** [comment]
The root of the issue is that the `.container` div within the `chip` has an interactive role (checkbox/radio/button) with a tabindex defined. Then inside of the `.container` div a `calcite-action` is rendered when the chip is closable. This fails the nested interactive elements requirement. To fix this, the close button needs to be outside of the other role so this will require some refactoring.

**Comment by geospatialem** [comment]
Adding the `design` label for additional considerations on the component, where the `role="button"` and nesting of the two buttons, the component and its close button, may need some tweaks altering the visual appearance and functionality of the component. 

_Note in testing by @isaacbraun_ When the component is within a `chip-group` the slotted `chip` and its close button are both keyboard accessible. However, when used as a standalone only the closable chip is keyboard accessible.

We may need to account for a similar functionality that is provided with `split-button`.

**Comment by ashetland** [comment]
Updated the description above to note updates to internal structure and hit targets.

**Comment by jcfranco** [comment]
This seems to be fixed by https://github.com/Esri/calcite-design-system/pull/14690. The nested element error is no longer reported by axe when using (5.2.0-next). cc @Esri/calcite-pes

**Comment by driskull** [comment]
@jcfranco I'm not sure that fixed the underlying issue. That seems to have just changed the role when closable but if a chip is closable and interactive or has a selectionMode the issue should still apply.

The chip component needs to be designed to not have nested interactive elements.
